src/libs/tracer/debugTraceCall.ts

```diff
@@ -51,57 +53,105 @@ export async function debugTraceCall(
-  if (!fast) return []
+  if (!fast) return { tokens: [], nfts: [] }
 
   const gasPrice =
     'gasPrice' in fast ? fast.gasPrice : fast.baseFeePerGas + fast.maxPriorityFeePerGas
 
   const params = getFunctionParams(account, op, accountState)
-  const results: string[] = await provider
-    .send('debug_traceCall', [
-      {
-        to: params.to,
-        value: toQuantity(params.value.toString()),
-        data: params.data,
-        from: params.from,
-        gasPrice: toQuantity(gasPrice.toString()),
-        gas: toQuantity(gasUsed.toString())
-      },
-      'latest',
-      {
-        tracer:
-          "{data: [], fault: function (log) {}, step: function (log) { if (log.op.toString() === 'LOG3') { this.data.push([ toHex(log.contract.getAddress()), '0x' + ('0000000000000000000000000000000000000000' + log.stack.peek(4).toString(16)).slice(-40)])}}, result: function () { return this.data }}",
-        enableMemory: false,
-        enableReturnData: true,
-        disableStorage: true,
-        stateOverrides: supportsStateOverride
-          ? {
-              [params.from]: {
-                balance: '0xffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffff'
-              }
+  const results: ({ erc: 20; address: string } | { erc: 721; address: string; tokenId: string })[] =
+    await provider
+      .send('debug_traceCall', [
+        {
+          to: params.to,
+          value: toQuantity(params.value.toString()),
+          data: params.data,
+          from: params.from,
+          gasPrice: toQuantity(gasPrice.toString()),
+          gas: toQuantity(gasUsed.toString())
+        },
+        'latest',
+        {
+          tracer: `{
+          data: [], 
+          fault: function (log) {}, 
+          step: function (log) { 
+            if (log.op.toString() === 'LOG3') {
+              this.data.push({
+                erc: 20,
+                address: toHex(log.contract.getAddress())
+              })
+            }
+            if (log.op.toString() === 'LOG4') { 
+              this.data.push({
+                erc: 721,
+                address: toHex(log.contract.getAddress()),
+                tokenId: '0x' + log.stack.peek(5).toString(16)
+              })
             }
-          : {}
-      }
-    ])
-    .catch(() => {
-      return [ZeroAddress]
-    })
-  const foundAddresses = [...new Set(results.flat(Infinity))]
+          }, 
+          result: function () { 
+            return this.data 
+          }
+        }`,
+
+          enableMemory: false,
+          enableReturnData: true,
+          disableStorage: true,
+          stateOverrides: supportsStateOverride
+            ? {
+                [params.from]: {
+                  balance: '0xffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffff'
+                }
+              }
+            : {}
+        }
+      ])
+      .catch((e) => {
+        console.log(e)
+        return [ZeroAddress]
+      })
```

Comment:
The return statement in `catch` is the old one. Maybe we should make it `return []` and check right after the promise if results are empty:
```
if (!results.length) return {tokens: [], nfts: []}
```